app/storage/metadata_parser.py

```diff
@@ -20,74 +19,69 @@ def id_generator():
-    'case_id': MetadataField(mandatory=False),
-    'case_ref': MetadataField(mandatory=False),
-    'account_service_url': MetadataField(mandatory=False),
+VALIDATORS = {
+    'date': iso_8601_date_parser,
+    'uuid': uuid_4_parser,
+    'string': lambda *args: None,
+    'object': lambda *args: None
 }
 
+MANDATORY_METADATA = {
+    'tx_id': 'uuid',
+}
 
-def parse_metadata(metadata_to_check):
-    parsed = {}
-    try:
-        for key, field in metadata_fields.items():
-            if key in metadata_to_check:
-                attr_value = metadata_to_check[key]
-                field.validate(attr_value)
-                logger.debug('parsing metadata', key=key, value=attr_value)
-            else:
-                logger.debug('generating metadata value', key=key)
-                attr_value = field.generate()
-
-            parsed[key] = attr_value
-    except (RuntimeError, ValueError, TypeError) as e:
-        logger.error('unable to parse metadata', exc_info=e)
-        raise InvalidTokenException('incorrect data in token')
-    return parsed
 
+def parse_metadata(decrypted_token, schema_metadata):
+    """
+    Extracts values from the JWT claims that are defined in the schema's metadata.
+    Ensures that the values adhere to the expected format.
+    :param decrypted_token:
+    :param schema_metadata:
+    :return: Validated claims
+    """
 
-def is_valid_metadata(metadata):
-    for key, field in metadata_fields.items():
-        if field.mandatory and key not in metadata:
-            return False, key
-    return True, ''
+    required_schema_metadata = _validate_mandatory_schema_metadata(decrypted_token, schema_metadata)
+    MANDATORY_METADATA.update(required_schema_metadata)
+    claims = decrypted_token.copy()
+    claims['tx_id'] = claims.get('tx_id', id_generator())  # Generate tx_id if not preset in claims
+
+    try:
+        for key, field in MANDATORY_METADATA.items():
+            if key in claims:
+                value = claims[key]
+                VALIDATORS[field](value)
+                logger.debug('parsing metadata', key=key, value=value)
+
+    except (RuntimeError, ValueError, TypeError) as error:
+        logger.error('unable to parse metadata', exc_info=error)
+        raise InvalidTokenException('incorrect data in token')
+    except KeyError as key_error:
+        error_msg = 'Invalid validator for schema metadata - {}'.format(key_error.args[0])
+        logger.error(error_msg, exc_info=key_error)
+        raise KeyError(error_msg)
+
+    return claims
+
+
+def _validate_mandatory_schema_metadata(metadata, schema_metadata):
+    """
+    Validate that JWT claims contain metadata defined within the schema
+    :param metadata:
+    :param schema_metadata:
+    :return:
+    """
+    for key in schema_metadata.keys():
+
+        if key == 'variant_flags':
+            # variant flags is empty by default
+            valid = 'variant_flags' in metadata
+        elif key == 'trad_as_or_ru_name':
```

Comment:
@MebinAbraham Just to confirm, we cant have both `variant_flags` and `ru_name` or `trad_as` together as mandatory metadata?